### DISCUSSION TITLE
using next-translate in Const

### DISCUSSION THREAD

**Discussion by vahidtakro** [vinissimus/next-translate#667] [Q&A] [answered]
Hello, I am a basic nextjs learner, so the template I am using is getting the header Nav Links from a file called HeaderNavLinks.js, which includes:
```
const headerNavLinks = [
  { href: '/blog', title: 'homepage' },
  { href: '/tags', title: 'tags' },
  { href: '/projects', title: 'projects' },
  { href: '/about', title: 'about' },
]

export default headerNavLinks
```

and the index page which refers to this is like this:
```
          <div className="flex items-center text-base leading-5">
            <div className="hidden sm:block">
              {headerNavLinks.map((link) => (
                <Link
                  key={link.title}
                  href={link.href}
                  className="p-1 font-medium text-gray-900 sm:p-4 dark:text-gray-100"
                >
                  {link.title}
                </Link>
              ))}
            </div>
```

my question is, how can I translate the titles in HeaderNavLinks.js?

**Comment by vspirandeli** [comment, marked as answer]
Hello vahidtakro. I use something loke this:
js file:
`const headerNavLinks = [
  { href: '/blog', title: 'namespace:homepage' }
]`
* Note that namespace should be your translate namespacefile.
and on page you can use:

`
import useTranslation from 'next-translate/useTranslation';
......

const { t } = useTranslation();
...
{headerNavLinks.map((link) => (
  <Link
    key={t(link.title)}
    href={link.href}
    ...
  >
    {t(link.title)}
  </Link>
  ....
`

**Comment by vahidtakro** [comment]
> Hello vahidtakro. I use something loke this:
> js file:
> `const headerNavLinks = [ { href: '/blog', title: 'namespace:homepage' } ]`
> 
> * Note that namespace should be your translate namespacefile.
>   and on page you can use:
> 
> `
> import useTranslation from 'next-translate/useTranslation';
> ......
> 
> const { t } = useTranslation();
> ...
> {headerNavLinks.map((link) => (
> 
> {t(link.title)} .... `

it works! thanks a lot!

can I internationalize my markdown based blog through this package?

**Comment by vspirandeli** [comment]
> > Hello vahidtakro. I use something loke this:
> > js file:
> > `const headerNavLinks = [ { href: '/blog', title: 'namespace:homepage' } ]`
> > 
> > * Note that namespace should be your translate namespacefile.
> >   and on page you can use:
> > 
> > `
> > import useTranslation from 'next-translate/useTranslation';
> > ......
> > const { t } = useTranslation();
> > ...
> > {headerNavLinks.map((link) => (
> > {t(link.title)} .... `
> 
> it works! thanks a lot!
> 
> can I internationalize my markdown based blog through this package?

If you use the example, i think it will works!!
